datatypes/data-value.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2018 gopcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package datatypes
+
+import (
+	"encoding/binary"
+	"time"
+
+	"github.com/wmnsk/gopcua/utils"
+)
+
+// DataValue is always preceded by a mask that indicates which fields are present in the stream.
+//
+// Specification: Part 6, 5.2.2.17
+type DataValue struct {
+	EncodingMask      byte
+	Value             *Variant
+	Status            uint32
+	SourceTimestamp   time.Time
+	SourcePicoSeconds uint16
+	ServerTimestamp   time.Time
+	ServerPicoSeconds uint16
+}
+
+// NewDataValue creates a new DataValue.
+func NewDataValue(hasValue, hasSrcTs, hasSrcPs, hasSvrTs, hasServerPc bool, v *Variant, status uint32, srcTs time.Time, srcPs uint16, svrTs time.Time, svrPs uint16) *DataValue {
```

Comment:
Can you remane `hasServerPc` as `hasSvrPs` for consistency?

Reply:
ah sorry about that